Added new kb article specifying-binding-cellcontenttemplate-datatable-xaml (#791)
telerik/maui-docs: knowledge-base/specifying-binding-cellcontenttemplate-datatable.md

---

> title: Specifying the Binding for a CellContentTemplate in Telerik .NET MAUI DataGrid with DataTable

Reviewer: I noticed that the "Telerik .NET MAUI" part gets appended by a template during the build. As the repetition is excessive, I suggest you skip the "Telerik .NET" part.
```suggestion
title: Specifying the CellContentTemplate Binding in a MAUI DataGrid with a DataTable
```

> description: This article explains how to specify the binding for a CellContentTemplate when using the .NET MAUI DataGrid DataTable.

Reviewer: ```suggestion
description: Learn how to specify the binding for a CellContentTemplate when using the .NET MAUI DataGrid DataTable.
```

type: how-to
page_title: Specifying the Binding for a CellContentTemplate DataTemplate
slug: specifying-binding-cellcontenttemplate-datatable
tags: DataGrid, DataTable, binding, template, dotnet, maui
res_type: kb
---
# Environment

| Version | Product | Author | 
| --- | --- | ---- | 
| 6.7.0 | Telerik UI for .NET MAUI DataGrid | [Dobrinka Yordanova](https://www.telerik.com/blogs/author/dobrinka-yordanova)| 

# Description

> When using a `CellContentTemplate` with the `RadDataGrid` that uses a `DataTable` as a source, you need to specify the binding for the template. This article provides an example of how to do this.

Reviewer: ```suggestion
When using a `CellContentTemplate` with a `RadDataGrid` that uses a `DataTable` as a source, you need to specify the binding for the template. This article provides an example of how to do this.
```

# Solution

To specify the binding for a `CellContentTemplate` follow these steps:

> 1. Define the RadDataGrid and set its `ItemsSource` property to the data source.

Reviewer: ```suggestion
1. Define the `RadDataGrid` and set its `ItemsSource` property to the data source.
```

public class DataTableViewModel : NotifyPropertyChangedBase
{
    private DataTable data;

    public DataTableViewModel()
    {
        this.Data = this.GetData();
    }

    public DataTable Data
    {
        get { return this.data; }
        set { this.UpdateValue(ref this.data, value); }
    }

    private DataTable GetData()
    {
        DataTable country = new DataTable();
        country.TableName = "Cities";
        country.Columns.Add("Id", typeof(int));
        country.Columns.Add("City", typeof(string));
        country.Columns.Add("Population", typeof(int));
        country.Columns.Add("Visited On", typeof(DateTime));
        country.Columns.Add("IsVisited", typeof(bool));

        country.Rows.Add(0, "Sofia", 2000000, new DateTime(2012, 10, 1), true);
        country.Rows.Add(1, "New York", 8419000, null, false);
        country.Rows.Add(2, "London", 8982000, new DateTime(2019, 02, 11), true);
        country.Rows.Add(3, "Los Angeles", 3967000, null, false);
        country.Rows.Add(4, "Budapest", 1765000, new DateTime(2013, 02, 1), true);
        country.Rows.Add(5, "Tokyo", 9375104, new DateTime(2015, 09, 1), true);

        return country;
    }
}
```

# Notes

> - The indexer syntax `[PropertyName]` is used to bind to a specific property in the DataRowView.

Reviewer: ```suggestion
- The indexer syntax `[PropertyName]` is used to bind to a specific property in the `DataRowView`.
```
